src/lib/core/hooks/study.ts

```diff
@@ -224,6 +227,10 @@ export function useFindEntityAndVariable(entities: StudyEntity[]) {
   );
 }
 
+export function useFlattenedCollectionVariables(entity: StudyEntity) {
```

Comment:
Can we rename this? I think "Flattened" is a little confusing.
```suggestion
export function useCollectionVariables(entity: StudyEntity) {
```